### PR TITLE
Added integration tests

### PR DESCRIPTION
**Pull request by StefanEnberg**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,3 +5,13 @@
 - async_stream_writer - takes the GRPC data chunks and writes them onto the GRPC stream.
 
 The reason for the division is to be able to utilize multiple CPU cores to process the different stages simultaneously.
+
+### Integration testing
+Goto src/test/integration and run:
+```
+ACCEPT_EULA=yes docker-compose up --build -d
```
Comment (wennmo) [inline README.md]: Usually we write `ACCEPT_EULA=<yes/no>` so people make an active decision to accept the eula and not just copy paste it

File: Dockerfile [hunk 2]
```diff
@@ -27,8 +27,8 @@ WORKDIR /usr/src/app
 # install maven dependency packages (keep in image)
 COPY pom.xml /usr/src/app
 
-RUN mvn install
 COPY src /usr/src/app/src
 
+RUN mvn install
```
Comment (wennmo) [inline Dockerfile]: Does `nvm install` has to be after copying the source? This way you won't make use of the docker cache layering.

File: src/test/integration/reload-runner/package.json [hunk 3]
```diff
@@ -8,13 +8,17 @@
     "start": "node index.js",
     "start:ci": "node index.js qix-engine",
     "lint": "eslint .",
-    "lint:fix": "eslint --fix ."
+    "lint:fix": "eslint --fix .",
+    "test": "mocha"
   },
   "dependencies": {
+    "chai": "^4.1.2",
```
Comment (wennmo) [inline src/test/integration/reload-runner/package.json]: lock dependencies

File: src/test/integration/reload-runner/type-check.js [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+  })
+  .then(() => {
+    console.log('Saving');
+    return app.doSave();
+  })
+  .then(() => {
+    //  type_small_int,type_integer,type_bigint,type_bit,type_numeric,type_double,type_real,type_decimal,type_boolean,type_date,type_time,type_timestamp,type_text,type_varchar,type_char
+    return app.getFieldDescription("type_date");
+//    console.log('Fetching Table sample');
+//    return app.getTableData(-1, 10000, true, 'AllTypes');
```
Comment (wennmo) [inline src/test/integration/reload-runner/type-check.js]: will there be some assertion/verification step added here?

File: Dockerfile (hunk 2 above)
Comment (StefanEnberg) [reply Dockerfile]: mvn builds to .proto-files and needs to be ran after they are copied

**Pull request merged**